### PR TITLE
Breaking: Switch UI to react

### PR DESCRIPTION
**Pull request by brafdlog**
- [x] Unify react wip flow into repo
- [x] Linting errors
- [x] Allow running react version side by side with vue version to allow merging before full feature parity
- [ ] Implement all flows
  - [x] Importers create/update/delete
  - [x] Importers enable/disable
  - [ ] Exporter settings
    - [x] Excel
    - [x] JSON
    - [x] Ynab
    - [ ] Google sheets
  - [x] General settings (num scraping days and show browser)
- [ ] Improve error indications in ynab exporter edit

Maybe in a separate PR:
- [ ] Improve UX to match design
- [ ] Improve logging to show full message including error
- [x] Solve ynab api issue and remove disabling of https check
- [ ] Error reporting (sentry)
- [ ] Analytics
- [ ] Download Chrome progress
- [ ] Remove vue
  - [ ] Switch to [electron-react-boilerplate](https://github.com/electron-react-boilerplate/electron-react-boilerplate)? 
  - [ ] Replace electron builder vue
  - [ ] Remove view files and build
- [ ] Check for updates flow
- [ ] Monorepo
  - [ ] Separate to worskpaces (yarn workspaces? [with lerna](https://medium.com/@jsilvax/a-workflow-guide-for-lerna-with-yarn-workspaces-60f97481149d))?
  - [ ] Solve react dependency issue and remove SKIP_PREFLIGHT_CHECK hack

### CODE REVIEW

**Comment by baruchiro** [comment]
I don't think we have to match to the design. I think we need to focus on supporting the same functionality as before.

Any improvments can be done by other PRs. We have to break the change.

**Comment by brafdlog** [comment]
Added the option to switch between the vew and react ui so we won't have to wait for full feature parity in order to merge.
I think I will add the ynab settings implementation and then want to merge (defaulting the ui to the old implementation but allowing to switch).
@baruchiro  WDYT?
(Ignore the redness of the screenshots, that is not how it looks)

image

image

**Comment by baruchiro** [comment]
@brafdlog Yes, I think it is the best option. Once it will be merged we will be able to do small changes, and also contributors will see their effect.

**Comment by brafdlog** [comment]
@baruchiro there are more things I want to add but since this allows the option of switching between old and new UI I don't think we should wait any longer.
Would love to get your feedback

**Review by baruchiro — APPROVED** [review]
OK, we miss a lot of UI features that exist in Vue, but we must to merge it. I hope we will have more time or more contributors.

- I re-merged the `electron` branch into this.
- What about keep the Vue UI as default?
- Did you tested the artifacts from the PR Validation action? We may test them and maybe ask users in our Discord.

**Review by baruchiro — CHANGES_REQUESTED** [review]
Please let's merge https://github.com/brafdlog/caspion/pull/378 before this, to give the "Electron Upgrade" a new release and release note for itself.

File: package.json
Comment (brafdlog) [reply package.json:19]: This is on purpose. It creates a separation of concerns between the UI code and electron code.
As we discussed a few times, this could easily become a cli, or use a different UI. Separating the package.json files allows us to have separate dependencies and will make this project more modular.

File: src/background.ts
Comment (brafdlog) [reply src/background.ts:46]: Good catch, missed this on the merge.
Fixed

**Review by baruchiro — APPROVED** [review]

File: package.json
Comment (baruchiro) [reply package.json:19]: Sure, I just think each package.json should be in a subfolder for its project, and maybe use `yarn workspaces` or something. But not now, we will consider it next time, because there is a pain in the separation.

**Comment by baruchiro** [comment]
>     * What about keep the Vue UI as default?

I know the e2e tests are not so useful, but they are what we have. I think keeping the Vue as default will ignore the tests problem, for now, see if you prefer to solve the issue or avoid it for now.

**Pull request merged**
